packages/react-components/src/components/ParticipantList.tsx

```diff
@@ -28,13 +37,32 @@ export type ParticipantListProps = {
   onRenderAvatar?: OnRenderAvatarCallback;
   /** Optional callback to render the context menu for each participant  */
   onParticipantRemove?: (userId: string) => void;
+  /** Optional callback to render custom menu items for each participant. */
+  onFetchParticipantMenuItems?: ParticipantMenuItemsCallback;
```

Comment:
I completely agree that `ParticipantMenuItemsCallback` is a good API at the composite layer.

Here though, it's a bit confusing.
 * Contoso can provide `onFetchParticipantMenuItems`, it will be used with the default implementation of `onRenderParticipant`.
 * Contoso can provide `onRenderParticipant`. In that case, `onFetchParticipantMenuItems` will be ignored.

 Same with other callbacks here. If `onRenderParticipant` is provided, `onRenderAvatar`, `onParticipantRemove` and `onFetchParticipantMenuItems` are all ignored. This is not obvious without looking at the implementation.

 Big question: Can we drop `onRenderParticipant` now that we have more targeted ways to customize this component?

Reply:
Valid concern there. This is going to be confusing only in components since we are not exposing onRenderParticipant via Composites.
To make it less confusing, we can improve the doc blocks.
TBH, I see this problem happening at multiple places throughout our components.
To handle this particular case, we could try and use the existing `onRenderParticipant` prop and pass it's attributes as defaults to the callback, but that will make the implementation really complex.

Reply:
I am not in favor of removing an `onRender` function as it takes away the ability of a dev to use a custom component. And there isn't any other way to do that for participantList